src/common/state/redis.c

```diff
@@ -157,12 +212,110 @@ void redis_object_table_lookup(table_callback_data *callback_data) {
   /* Call redis asynchronously */
   redisAsyncCommand(db->context, redis_object_table_get_entry,
                     (void *) callback_data->timer_id, "SMEMBERS obj:%b",
-                    &callback_data->id.id[0], UNIQUE_ID_SIZE);
+                    &callback_data->id.id[0], (size_t) UNIQUE_ID_SIZE);
   if (db->context->err) {
     LOG_REDIS_ERR(db->context, "error in object_table lookup");
   }
 }
 
+void redis_result_table_add_callback(redisAsyncContext *c,
+                                     void *r,
+                                     void *privdata) {
+  REDIS_CALLBACK_HEADER(db, callback_data, r)
+  redisReply *reply = r;
+  CHECK(reply->type == REDIS_REPLY_STATUS ||
+        reply->type == REDIS_REPLY_INTEGER);
+  if (callback_data->done_callback) {
+    result_table_done_callback done_callback = callback_data->done_callback;
+    done_callback(callback_data->id, callback_data->user_context);
+  }
+  task_id *task_id = callback_data->data;
+  free(task_id);
+  destroy_timer_callback(db->loop, callback_data);
+}
+
+void redis_result_table_add(table_callback_data *callback_data) {
+  CHECK(callback_data);
+  db_handle *db = callback_data->db_handle;
+  task_id *result_task_id = (task_id *) callback_data->data;
+  /* Add the result entry to the result table. */
+  int status =
+      redisAsyncCommand(db->context, redis_result_table_add_callback,
+                        (void *) callback_data->timer_id, "SET result:%b %b",
+                        callback_data->id.id, (size_t) UNIQUE_ID_SIZE,
+                        (*result_task_id).id, (size_t) UNIQUE_ID_SIZE);
+  if ((status == REDIS_ERR) || db->context->err) {
+    LOG_REDIS_ERR(db->context, "Error in result table add");
+  }
+}
+
+void redis_result_table_lookup_task_callback(redisAsyncContext *c,
+                                             void *r,
+                                             void *privdata) {
+  REDIS_CALLBACK_HEADER(db, callback_data, r)
+  redisReply *reply = r;
+  /* Check that we received a Redis hashmap. */
+  if (reply->type != REDIS_REPLY_ARRAY) {
+    LOG_ERR("Expected Redis array, received type %d %s", reply->type,
+            reply->str);
+    exit(-1);
```

Comment:
Ah, this should probably be `CHECK`.

`CHECK` is fatal and `LOG_ERR` is non-fatal errors. We could probably name these better.